docs/blocks-storage/querier.md

```diff
@@ -1341,6 +1341,51 @@ blocks_storage:
     # CLI flag: -blocks-storage.bucket-store.series-batch-size
     [series_batch_size: <int> | default = 10000]
 
+    token_bucket_limiter:
+      # Whether token bucket limiter is enabled
+      # CLI flag: -blocks-storage.bucket-store.token-bucket-limiter.enabled
+      [enabled: <boolean> | default = false]
+
+      # Whether the token bucket limiter is in dry run mode
+      # CLI flag: -blocks-storage.bucket-store.token-bucket-limiter.dry-run
+      [dry_run: <boolean> | default = false]
+
+      # Pod token bucket size
+      # CLI flag: -blocks-storage.bucket-store.token-bucket-limiter.pod-token-bucket-size
+      [pod_token_bucket_size: <int> | default = 859832320]
+
+      # User token bucket size
+      # CLI flag: -blocks-storage.bucket-store.token-bucket-limiter.user-token-bucket-size
+      [user_token_bucket_size: <int> | default = 644874240]
+
+      # Request token bucket size
+      # CLI flag: -blocks-storage.bucket-store.token-bucket-limiter.request-token-bucket-size
+      [request_token_bucket_size: <int> | default = 4194304]
+
+      # Multiplication factor used for fetched postings token
+      # CLI flag: -blocks-storage.bucket-store.token-bucket-limiter.fetched-postings-token-factor
+      [fetched_postings_token_factor: <float> | default = 2]
+
+      # Multiplication factor used for touched postings token
+      # CLI flag: -blocks-storage.bucket-store.token-bucket-limiter.touched-postings-token-factor
+      [touched_postings_token_factor: <float> | default = 2]
+
+      # Multiplication factor used for fetched series token
+      # CLI flag: -blocks-storage.bucket-store.token-bucket-limiter.fetched-series-token-factor
+      [fetched_series_token_factor: <float> | default = 2.5]
+
+      # Multiplication factor used for touched series token
+      # CLI flag: -blocks-storage.bucket-store.token-bucket-limiter.touched-series-token-factor
+      [touched_series_token_factor: <float> | default = 10]
```

Comment:
Is there a reason why touched series has more weight than fetched series?

Reply:
I've performed load test to come up with these default values. I used caches to control the amount of fetched vs touched data, and it was found that touched series had most impact to CPU.

Reply:
Umm, can we share the tests you ran so that Cortex users understand the reason?
I am really 100% convinced as the result makes me doubt that why we need to enable series caching then if fetching series from S3 is better than fetching from cache.

Reply:
So I've performed test once more to verify, and it seems like fetching from S3 is indeed better for CPU, although it's worse for bandwidth.

<img width="1707" alt="Screenshot 2024-06-25 at 9 15 34 AM" src="https://github.com/cortexproject/cortex/assets/18124240/80756994-6824-4926-8cc3-c93a936c42ed">

I've disabled cache with the following cli config:

```
      exec cortex \
        -target=store-gateway \
        -config.expand-env \
        -blocks-storage.bucket-store.index-cache.inmemory.max-size-bytes=1073741824 \
        -blocks-storage.bucket-store.index-cache.backend=inmemory \
        -blocks-storage.bucket-store.index-cache.inmemory.enabled-items=ExpandedPostings \
        -config.file=/etc/config/config.yml
```

Reply:
I've uploaded test results for postings and chunks as well, added to the PR description.
I've also revised the default values, such that we use data bytes touched only for the token bucket as it seems like data bytes fetched don't seem to impact CPU or memory.

Reply:
Thanks Justin. I think I am good now. For every series SG fetches it also increments `series_touched` as well so fetching series from S3 will use more token than fetching from cache. This behavior is ok to me.